source/_posts/2018-01-08-how-to-quickly-deploy-a-vuejs-app-to-heroku.markdown

```diff
@@ -0,0 +1,163 @@
+
+* Write a minimal NodeJS web server using Express
+* Build the assets locally
+* Add the `dist` folder to the Git repository, so it is included when pushing to Heroku
+
+What I didn’t like of this solution was the need to build locally the site and check in the changes within the `dist` folder. I wanted to have this step handled by Heroku when pushing a new version of my application.
+
+## Our solution
+Let's assume we have a VueJS project generated using `vue-cli` with the `webpack` template. Just to be clear, the project was created using the following command:
+
+```bash
+vue init webpack <YOUR-PROJECT-NAME-HERE>
+``` 
+
+Of course, we also need a Heroku account and a new application created there. Heroku will use the NodeJS buildpack because our project contains a `package.json` in the root folder.
+
+### Step 1: Add a minimal NodeJS server
+
+This is a step borrowed from the mentioned blog posts. We have to add a `server.js` file in the project’s root folder with the following code:
+
+```js
+const express = require('express');
+const path = require('path');
+const serveStatic = require('serve-static');
+
+let app = express();
+app.use(serveStatic(__dirname + "/dist"));
+
+const port = process.env.PORT || 5000;
+app.listen(port, () => {
+  console.log('Listening on port ' + port)
+});
+```
+
+Since this code uses Express, we need to add this dependency to our project:
+
+```bash
+npm install express --save
+```
+
+You can test this server locally by running the following commands:
+
+```bash
+npm run build
+node server.js
+```
+
+### Step 2: Setup package.json scripts
+
+We need to tweak the `scripts` section in the `package.json`. If package file provided by the Vue Webpack template was not modified, it should include two important tasks, `start` and `build`:
+
+```json
+"scripts": {
+  ...
+  "start": "npm run dev",
+  ...
+  "build": "node build/build.js"
+},
+```
+
+By default, the `start` script will be executed by Heroku to start the server. For this reason, we will change the command associated to `start` to run our custom server script:
+
+```json
+"scripts": {
+  ...
+  "start": "node server.js",
+  ...
+},
+```
+
+Please note you can not use `npm run start` anymore to run the development server in your computer. I just decided to use `npm run dev` directly but you could add a new entry in the `scripts` section with an alias for that.
+
+Now, we have to add something to make sure that the `dist` folder is built in our Heroku instance, once the code is deployed there. Otherwise, the server script is not going to work properly. We will use an special script called `heroku-postbuild` which is documented [here](https://devcenter.heroku.com/articles/nodejs-support#heroku-specific-build-steps). The idea is to build the site using this special hook, so let’s add it to our `package.json`:
```

Comment:
use an special -> use a special

Reply:
How about "Once the code is deployed, we have to add something to make sure that the `dist` folder is built in our Heroku instance"?

Reply:
the punctuation is sort of confusing there...re-writing like @mariiio said would do, or if you don't want to re-write it I'd just change the pauses to

`Now we have to add something to make sure that the `dist` folder is built in our Heroku instance once the code is deployed there, otherwise the server script is not going to work properly. `

Reply:
I think that @mariiio is changing a bit the meaning of the phrase but I see the problem.

I'm changing to `We still have to add something to make sure that the `dist` folder is built in our Heroku instance every time the code is deployed, otherwise the server script is not going to work properly`